src/Temporalio/Client/Configuration/ClientConfigProfile.cs

```diff
@@ -0,0 +1,188 @@
+        /// Initializes a new instance of the <see cref="ClientConfigProfile"/> class.
+        /// </summary>
+        /// <param name="address">Client address.</param>
+        /// <param name="namespace">Client namespace.</param>
+        /// <param name="apiKey">Client API key.</param>
+        /// <param name="tls">TLS configuration.</param>
+        /// <param name="grpcMeta">gRPC metadata.</param>
+        public ClientConfigProfile(
+            string? address = null,
+            string? @namespace = null,
+            string? apiKey = null,
+            ClientConfigTls? tls = null,
+            IDictionary<string, string>? grpcMeta = null)
+        {
+            Address = address;
+            Namespace = @namespace;
+            ApiKey = apiKey;
+            Tls = tls;
+            if (grpcMeta != null)
+            {
+                GrpcMeta = new Dictionary<string, string>(grpcMeta);
+            }
+        }
+
+        /// <summary>
+        /// Gets the client address.
+        /// </summary>
+        public string? Address { get; private set; }
+
+        /// <summary>
+        /// Gets the client namespace.
+        /// </summary>
+        public string? Namespace { get; private set; }
+
+        /// <summary>
+        /// Gets the client API key.
+        /// </summary>
+        public string? ApiKey { get; private set; }
+
+        /// <summary>
+        /// Gets the TLS configuration.
+        /// </summary>
+        public ClientConfigTls? Tls { get; private set; }
+
+        /// <summary>
+        /// Gets the gRPC metadata.
+        /// </summary>
+        public IDictionary<string, string>? GrpcMeta { get; private set; }
+
+        /// <summary>
+        /// Loads a specific profile with environment variable overrides.
+        /// </summary>
+        /// <param name="profile">Name of the profile to load. Defaults to "default" if not specified.</param>
+        /// <param name="configSource">The data source to load from.</param>
+        /// <param name="disableFile">If true, do not load from file (only from environment).</param>
+        /// <param name="disableEnv">If true, disable environment variable overrides.</param>
+        /// <param name="configFileStrict">If true, fail if configuration file is invalid.</param>
+        /// <param name="overrideEnvVars">Environment variables to use, or null to use system environment.</param>
+        /// <returns>The loaded profile.</returns>
+        public static ClientConfigProfile Load(
+            string? profile = null,
+            DataSource? configSource = null,
+            bool disableFile = false,
+            bool disableEnv = false,
+            bool configFileStrict = false,
+            Dictionary<string, string>? overrideEnvVars = null)
+        {
+            if (disableFile && disableEnv)
+            {
+                throw new InvalidOperationException("Cannot disable both file and environment sources - at least one must be enabled");
```

Comment:
General question, what happens if I provide a `configSource` with some data and set `disableFile` as `true`? Does it disregard my `configSource` parameter? Or if not, then it is reasonable to have `disableFile` and `disableEnv` when there is a `configSource` right?